### PR TITLE
FormatterType Support SF3

### PR DESCRIPTION
**Pull request by rafaelcalleja**
add to #145 sonata_formatter_type support in SF3

### CODE REVIEW

**Comment by OskarStark** [comment]
LGTM

**Comment by greg0ire** [comment]
And the commit adds tests… great work!

File: Tests/Form/Widget/BaseWidgetTest.php [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function tearDown()
+    {
+        parent::tearDown();
+
+        $this->extension = null;
```
Comment (core23) [inline Tests/Form/Widget/BaseWidgetTest.php]: IMO, we don't need this

File: Tests/Form/Widget/BaseWidgetTest.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+        $renderer = new TwigRenderer($rendererEngine, $this->getMock($csrfManagerClass));
+
+        $this->extension = new FormExtension($renderer);
+
+        $twigPaths = array(__DIR__.'/../../../Resources/views/Form');
+
+        //this is ugly workaround for different build strategies and, possibly,
+        //different TwigBridge installation directories
+        if (is_dir(__DIR__.'/../../../vendor/symfony/twig-bridge/Resources/views/Form')) {
```
Comment (core23) [inline Tests/Form/Widget/BaseWidgetTest.php]: Could we move this to a seperate method? So we can use this for other tests too

Comment (greg0ire) [inline Tests/Form/Widget/BaseWidgetTest.php]: What do you mean? It already is in a public method…

File: Tests/Form/Widget/BaseWidgetTest.php (hunk 1 above)
Comment (dnoise) [inline Tests/Form/Widget/BaseWidgetTest.php]: done

**Comment by dnoise** [comment]
I've upgraded this PR with master because the Test was failing but the upgrade doesn't fix the Test.

Thank to the failed test I guess have detected a issue about versions of egeloen/ckeditor-bundle from the last merge #110 .

Too In my project i can see:
`The service "sonata.formatter.twig.ck_editor" has a dependency on a non-existent service "ivory_ck_editor.templating.helper".`

The service ivory_ck_editor.templating.helper was renamed in version 2.4 but composer.json is allowing version  2.2 
`"egeloen/ckeditor-bundle": "^2.2|^3.0"`

You can check it from [this commit](https://github.com/egeloen/IvoryCKEditorBundle/commit/5e8fc1094cffb3072373d72899a240a5958378ff)

To fix it and the test too simply need change composer.json to

`"egeloen/ckeditor-bundle": "~2.4|^3.0",`

maybe this change would be in a new PR or Issue?

**Comment by greg0ire** [comment]
> maybe this change would be in a new PR or Issue?

I think it would. Can you do it? @Th3Mouk can you comment on this?

**Comment by dnoise** [comment]
The firt version of egeloen/ckeditor-bundle with support for symfony/form:~3.0 is [4.0](https://github.com/egeloen/IvoryCKEditorBundle/blob/4.0.0/composer.json)

I guess we can support sf3 with something like this
"egeloen/ckeditor-bundle": "~2.4|^3.0|~4.0",

**Comment by Th3Mouk** [comment]
I've started to develop the upgrade in october and there's not 4.x releases, i agree that the composer.json must be upgrade from `^2.2` to `^2.5` because when i've tested i was in `2.5.2`.

Il will make the upgrade of composer.

The ^4.0 upgrade seems really simple, @dnoise would you do the PR ?

**Comment by Th3Mouk** [comment]
I've check it's ok for 4.x support there no BC, i've made PR to https://github.com/sonata-project/SonataFormatterBundle/pull/148

File: Tests/Form/Widget/BaseWidgetTest.php (hunk 2 above)
Comment (core23) [inline Tests/Form/Widget/BaseWidgetTest.php]: Can't we use a global WidgetTest class @greg0ire ? We have an existing here https://github.com/sonata-project/SonataAdminBundle/blob/master/Tests/Form/Widget/BaseWidgetTest.php

Comment (greg0ire) [inline Tests/Form/Widget/BaseWidgetTest.php]: Formatter depends on Core, so I think we could try to factor out something into core. Maybe let it mature into something generic enough first, b/c tests must stay easy to write.

**Comment by OskarStark** [comment]
can we merge this @Th3Mouk

**Comment by Th3Mouk** [comment]
@OskarStark I can't pronounce on the testing part, that must be extract on a base class apparently.

@rafaelcalleja could you rebase please ? Tests are red actually

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**